api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/authenticate/ForgotPasswordController.java

```diff
@@ -0,0 +1,416 @@
+    private CaptchaImplementation captchaImpl;
+
+    @Override
+    protected ResponseValues processRequest(VitroRequest vreq) throws Exception {
+        // Random time interval sleep so attacker can't calculate whether provided email is bound to an account or not
+        sleepForRandomTime();
+
+        captchaImpl = CaptchaServiceBean.getCaptchaImpl();
+
+        Map<String, Object> dataContext = new HashMap<>();
+        setCommonValues(dataContext, vreq);
+        CaptchaServiceBean.addCaptchaRelatedFieldsToPageContext(dataContext);
+        boolean isEnabled = isFunctionalityEnabled();
+        dataContext.put("isEnabled", isEnabled);
+
+        // Handle GET request (display the form) or print error message if functionality is disabled
+        if (!isEnabled || vreq.getMethod().equalsIgnoreCase("GET")) {
+            return showForm(dataContext);
+        }
+
+        return handlePostRequest(vreq, dataContext);
+    }
+
+    /**
+     * Handles a POST request for password reset. Validates the captcha, checks for spam mitigation,
+     * processes the password change request, and notifies the user.
+     *
+     * @param vreq        The Vitro request object.
+     * @param dataContext A map containing additional data context for processing the request.
+     * @return A response containing the appropriate values for rendering the result.
+     */
+    private ResponseValues handlePostRequest(VitroRequest vreq, Map<String, Object> dataContext) {
+        UserAccountsDao userAccountsDao = constructUserAccountsDao(vreq);
+        I18nBundle i18n = I18n.bundle(vreq);
+
+        String email = vreq.getParameter("email");
+
+        if (!captchaIsValid(vreq)) {
+            dataContext.put("wrongCaptcha", true);
+            dataContext.put("emailValue", email);
+            return showForm(dataContext);
+        }
+
+        dataContext.put("showPasswordChangeForm", false);
+
+        PasswordChangeRequestSpamMitigationResponse mitigationResponse =
+            PasswordChangeRequestSpamMitigation.isPasswordResetRequestable(email);
+        if (!mitigationResponse.getCanBeRequested()) {
+            dataContext.put("message",
+                i18n.text("password_reset_too_many_requests", mitigationResponse.getNextRequestAvailableAtDate(),
+                    mitigationResponse.getNextRequestAvailableAtTime()));
+            return new TemplateResponseValues(TEMPLATE_NAME, dataContext);
+        }
+
+        Optional<UserAccount> userAccountOptional = getAccountForInternalAuth(email, vreq);
+        if (userAccountOptional.isPresent()) {
+            requestPasswordChange(userAccountOptional.get(), userAccountsDao);
+            notifyUser(userAccountOptional.get(), i18n, vreq);
+        } else {
+            log.info("User tried to reset password with an unassociated email: " + email);
+        }
+
+        PasswordChangeRequestSpamMitigation.requestSuccessfullyHandledAndUserIsNotified(email);
```

Comment:
Even if user doesn't exists? I suppose method name is misleading.

Reply:
Renamed